server/src/main/java/com/google/musicanalysis/api/youtube/YoutubeRequest.java

```diff
@@ -38,6 +39,12 @@ public String getResult() throws MalformedURLException, IOException {
 
     URL url = buildUrl();
 
+    HttpURLConnection con = (HttpURLConnection) url.openConnection();
+    int response = con.getResponseCode();
+    if (response > 400) {
```

Comment:
check if response != 200 instead, a 301 response won't be very useful to you either